base/tps/src/org/dogtagpki/server/tps/cms/KRARemoteRequestHandler.java

```diff
@@ -66,14 +66,21 @@ public KRAServerSideKeyGenResponse serverSideKeyGen(
             String userid,
             String sDesKey,
             String sAesKey,
-            boolean archive)
+            boolean archive,
+            String aesKeyWrapAlg)
             throws EBaseException {
 
         CMS.debug("KRARemoteRequestHandler: serverSideKeyGen(): begins.");
         if (cuid == null || userid == null || sDesKey == null) {
             throw new EBaseException("KRARemoteRequestHandler: serverSideKeyGen(): input parameter null.");
         }
 
+        String aesWrapAlg = aesKeyWrapAlg;
+
+        if(aesWrapAlg == null || aesWrapAlg.length() < 3) {
```

Comment:
why "<3"?  What if it's "abcd"?

Reply:
I think I did that cause I know I"m supporting only "KWP and "CBC" at this point, but I can make it > 0

Reply:
`OK upon further review I did this to make sure it was at least 3 chars for KWP or CBC. If not, then we would just take the well known default of KWP. I guess I could just make it ( != 3 ). Anyway I will have something for the next submission.

Reply:
Done.

Reply:
How about simply say: if ((aesWrapAlg == null) || !aesWrapAlg.equalsIgnoreCase("cbc")) aesWrapAlg = "KWP";